app/settings/data-export/hdx-export.controller.js

```diff
@@ -0,0 +1,149 @@
+module.exports = [
+    '$scope',
+    '$rootScope',
+    'Features',
+    '$location',
+    'HxlExport',
+    '_',
+    'LoadingProgress',
+function (
+    $scope,
+    $rootScope,
+    Features,
+    $location,
+    HxlExport,
+    _,
+    LoadingProgress
+) {
+    $scope.selectAttribute = selectHxlAttribute;
+    $scope.addAnother = addAnother;
+    $scope.range = range;
+    $scope.selectTag = selectTag;
+    $scope.exportData = exportData;
+    $scope.selectAll = selectAll;
+    $scope.isLoading = LoadingProgress.getLoadingState;
+    $scope.getSelectedFields = getSelectedFields;
+    $scope.hxlAttributeSelected = hxlAttributeSelected;
+
+    // Change layout class
+    $rootScope.setLayout('layout-c');
+    // Change mode
+    $scope.$emit('event:mode:change', 'settings');
+
+    // Check if hxl-feature is enabled
+    Features.loadFeatures().then(function () {
+        $scope.hxlEnabled = Features.isFeatureEnabled('hxl');
+        // Redirect to home if not enabled
+        if (!$scope.hxlEnabled) {
+            return $location.path('/');
+        }
+    });
+
+    activate();
+
+    function activate() {
+        HxlExport.getFormsWithTags().then((formsWithTags)=> {
+            $scope.forms = formsWithTags;
+        });
+    }
+
+    function range(attribute) {
+        if (!attribute.nbAttributes) {
+            attribute.nbAttributes = 1;
+        }
+        return _.range(attribute.nbAttributes);
+    }
+
+    function addAnother(attribute) {
+        attribute.nbAttributes++;
+    }
+
+    function hxlAttributeSelected(hxlAttribute, formAttribute, index) {
+        let disabled = false;
+        // hack to not disable the value in the current dropdown
+        let selectedHxlAttributes = angular.copy(formAttribute.selectedHxlAttributes);
+        delete selectedHxlAttributes[index];
+        if (_.findWhere(selectedHxlAttributes, {id: hxlAttribute.id})) {
+            disabled = true;
+        }
+        return disabled;
+    }
+
+    function selectAll(form) {
+        if (form.selected) {
+            _.each(form.attributes, (attribute) => {
+                attribute.selected = [attribute.id];
```

Comment:
Hey, do we need to use a for each here or can we use a `map` instead?  
ie : 
```
form.attributes =  form_attributes.map(attribute => {
    attribute.selected = [attribute.id];
    return attribute;
}
```

feel free to ignore if too nitpicky/can't do it now, it's just a suggestion since I'd prefer it if we use `map` when we can and when changing objects in the array. It's also faster than foreach 🎉 and  allows for chaining if you return it which is handy.

Reply:
Fixed 🎉